lib/smack/Prelude.cpp

```diff
@@ -280,39 +290,88 @@ std::string getBytewisePointerStorageStr() {
   return s.str();
 }
 
-std::string Prelude::getPrelude() {
-  const std::vector<unsigned> INTEGER_SIZES {
-    1, 5, 6, 8, 16, 24, 32, 40, 48, 56, 64, 80, 88, 96, 128, 160, 256
-  };
+const std::vector<unsigned> INTEGER_SIZES {
+  1, 5, 6, 8, 16, 24, 32, 40, 48, 56, 64, 80, 88, 96, 128, 160, 256
+};
 
-  const std::vector<unsigned> REF_CONSTANTS {
-    0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 1024
-  };
+const std::vector<unsigned> REF_CONSTANTS {
+  0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 1024
+};
 
-  const std::vector<std::string> BINARY_OPS {
+void Prelude::generateBinaryOps(std::stringstream& s, const unsigned size) {
+  static const std::vector<std::string> BINARY_OPS {
     "add", "sub", "mul", "sdiv", "smod", "srem", "udiv",
-    "urem", "shl", "lshr", "ashr", "and", "or", "xor", "nand"
+      "urem", "shl", "lshr", "ashr", "and", "or", "xor", "nand"
   };
+  static const std::vector<BinExpr::Binary> BINARY_OPS_BODY {
+    BinExpr::Plus, BinExpr::Minus, BinExpr::Times
+  };
+  for (size_t i = 0; i < BINARY_OPS.size(); ++i) {
+    const std::string &op = BINARY_OPS[i];
+    s << bvbuiltinBinaryOp(size, "bv" + op, "$" + op) << "\n";
+    if (i <= 2) {
```

Comment:
It is weird that index decides into which category the op falls into. Could we change that into something else?
You could maybe split them into multiple vectors. Or they could be `<string,type>`pairs in the same vector.

Reply:
Yes, I'll fix it.